Monika After Story/game/zz_weather.rpy

```diff
@@ -120,6 +163,42 @@ init -20 python in mas_weather:
 #            or store.persistent._mas_weather_snow_happened
 #        )
 
+    def weatherProgress():
+        """
+        Runs a roll on mas_shouldRain() to pick a new weather to change to after a time between half an hour - one and a half hour
+
+        RETURNS:
+            - True or false on whether or not to call spaceroom
+        """
+        import datetime
+        import random
+
+        #If it shouldn't rain today, or the player forced weather, then we do nothing
+        if not shouldRainToday() or mas_force_weather:
+            return False
+
+        #Otherwise we do stuff
+        global weather_change_time
+        #Set a time for startup
+        if not weather_change_time:
+            weather_change_time = datetime.datetime.now() + datetime.timedelta(0,random.randint(1800,5400))
+
+        elif weather_change_time < datetime.datetime.now():
+            #Need to set a new check time
+            weather_change_time = datetime.datetime.now() + datetime.timedelta(0,random.randint(1800,5400))
+
+            #Change weather
+            new_weather = store.mas_shouldRain()
+            if new_weather is not None and new_weather.prompt != store.mas_current_weather.prompt:
```

Comment:
weather objects have `weather_id` property. you should use that here instead of prompt.